### ISSUE TITLE
process_a11y.py script is not factoring in domains with no errors but it should

### ISSUE THREAD

**Issue by gbinal** [18F/domain-scan#103]
In step 6 of the a11y [scanning process](https://github.com/18F/domain-scan/wiki/a11y-scanning-process), the [process_a11y.py script](https://github.com/18F/domain-scan/blob/make-lambda-optional/scripts/a11y/process_a11y.py) is not factoring in domains that have no errors since it is just building from the results of the a11y.csv file generated in step 5, however it needs to.  

Imagine an executive branch agency with three active, non-redirecting domains.  After step 5 completes, there are only error results for two domains (either because the third domain did not scan successfully or because no errors were detected).  The problem is that step 6 computes based on the a11y.csv file of individual error results and does not factor in the total domain set that it should be considering.

**Comment by gbinal** [comment]
To elaborate, when the `process_a11y.py` script runs, it needs to do the following.  

When creating the `agencies.json` file: 
* `Average Errors per Page` should be computed by `[Sum of Errors]/[Number of Domains The Agency Has In Domains.csv]` instead of what is currently the case, `[Sum of Errors]/[Number of Domains The Agency Has In A11y.csv]`.  
* If an agency is represented in `domains.csv` but not in `a11y.csv`, an entry for it should still be created, with `0` for each of the error fields.    
  
When creating the `domains.json` file: 
* If a domain is represented in `domains.csv` but not in `a11y.csv`, an entry for it should still be created, with `0` for each of the error fields.

**Comment by gbinal** [comment]
@micahsaul's good work has resolved this so I'll close now.
